cpp/src/arrow/chunk_resolver.h

```diff
@@ -20,13 +20,16 @@
 #include <atomic>
 #include <cassert>
 #include <cstdint>
+#include <limits>
 #include <vector>
 
 #include "arrow/type_fwd.h"
 #include "arrow/util/macros.h"
 
 namespace arrow::internal {
```

Comment:
Since everything here is internal, do you think we can rename this header file to `chunk_resolver_internal.h`? Not necessarily in this PR...

Reply:
#34535